Formatter/Pool.php

```diff
@@ -34,15 +36,43 @@ class Pool
     protected $logger;
 
     /**
-     * @param LoggerInterface|null $logger
-     * @param string|null          $defaultFormatter
+     * @param string|null $defaultFormatter
      */
-    public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
+    public function __construct($defaultFormatter = null)
     {
-        $this->logger = $logger;
+        // NEXT_MAJOR: remove this block
+        if (func_num_args() == 2) {
+            list($logger, $defaultFormatter) = func_get_args();
+            $this->logger = $logger;
+            $this->defaultFormatter = $defaultFormatter;
 
-        // TODO: This should become a required first parameter when the major version is changed
+            return;
+        }
+        if (func_num_args() == 1 && $defaultFormatter instanceof LoggerInterface) {
+            $this->logger = $defaultFormatter;
+            @trigger_error(sprintf(
+                'Providing a logger to %s through the constructor is deprecated since 3.x'.
+                ' and we no longer be possible in 4.0'.
+                ' This argument should be provided through the setLogger() method.',
+                __METHOD__
+            ), E_USER_DEPRECATED);
+
+            return;
+        }
         $this->defaultFormatter = $defaultFormatter;
+        // NEXT_MAJOR: make defaultFormatter required
+        if (is_null($defaultFormatter)) {
+            @trigger_error(sprintf(
+                'Not providing the defaultFormatter argument to %s is deprecated since 3.x.'.
+                ' This argument will become mandatory in 4.0.',
+                __METHOD__
+            ), E_USER_DEPRECATED);
+        }
```

Comment:
@sagikazarmark : I'm done with the `LoggerAwareInterface` implementation (I think). Looks good?